### PR TITLE
nimon file added

### PR DESCRIPTION
**Pull request by akshaym99**
Signed-off-by: akshaym99 <akshay.s.mankar@seagate.com>

## Problem Statement

-   Problem statement

## Design

-   For Bug, Describe the fix here.
-   For new test addition post the link for design

## Coding

   Checklist for Author

-   [ ]  Coding conventions are followed and code is formatted (e.g. using pycharm in-built formatter)

## Testing

  Checklist for Author

-   [ ]  New/Affected tests are executed
-   [ ]  Attach test execution logs

## Review 

  Checklist for Author

-   [ ]  JIRA number/GitHub Issue added to PR
-   [ ]  PR is self reviewed
-   [ ]  Jira and state/status is updated and JIRA is updated with PR link
-   [ ]  Check if the description is clear and explained

## Documentation

  Checklist for Author

-   [ ]  Changes done to ReadMe

### CODE REVIEW

File: src/commons/utils/nimon_resource_util.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            worker_node = resp[1].strip().split("\n")[1:]
+            LOGGER.info("worker nodes: %s", str(worker_node))
+            server_nodes.extend(worker_node)
+    if action == "start":
+        resp = is_package_installed_local("unzip")
+        LOGGER.debug("Local response: %s", str(resp))
+        if not resp[0]:
+            resp = run_local_cmd(cm_cmd.YUM_UNZIP)
+            LOGGER.info("Local response: %s", str(resp))
+        resp = is_package_installed_local("nimon")
```
Comment (ketanarlulkar) [inline src/commons/utils/nimon_resource_util.py:62]: nimon or nmon?

File: src/commons/utils/nimon_resource_util.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        LOGGER.debug("Local response: %s", str(resp))
+        if not resp[0]:
+            resp = run_local_cmd(cm_cmd.CMD_WGET_NIMON)
+            LOGGER.info("Local response: %s", str(resp))
+            resp = run_local_cmd(cm_cmd.UNZIP_NIMON)
+            LOGGER.info("Local response: %s", str(resp))
```
Comment (ketanarlulkar) [inline src/commons/utils/nimon_resource_util.py]: Better to log response in `run_local_command` instead here after each call. Also log command and response both, not just response.

File: src/commons/utils/nimon_resource_util.py (hunk 1 above)
Comment (akshaym99) [reply src/commons/utils/nimon_resource_util.py:62]: nimon is different than nmon.
It's supports for real time analysis.

File: src/commons/utils/nimon_resource_util.py (hunk 2 above)
Comment (akshaym99) [reply src/commons/utils/nimon_resource_util.py]: Logging inside run_local_cmd (command also logging inside).

File: src/commons/constants.py [hunk 3]
```diff
@@ -57,7 +57,7 @@
 ROOT = "corio"  # root logger name.
 MIN_DURATION = 20  # Minimum execution duration in seconds.
 DT_STRING = datetime.now().strftime("%d_%m_%Y_%H_%M_%S")
-
+# nmon installation
 CMD_YUM_NMON = "yum install -y nmon"
```
Comment (DhobaleVishal) [inline src/commons/constants.py]: Group the related commands.

File: src/commons/utils/corio_utils.py [hunk 4]
```diff
@@ -384,3 +397,17 @@ def list_dirs(self, remote_path: str) -> list:
...
             self.disconnect()
+
+    def is_package_installed(self, package_name):
+        """
+        check package is installed or not.
+
+        :param package_name: package name to check
+        """
+        self.connect()
+        resp = self.execute_command("rpm -qa | grep {}".format(package_name))
```
Comment (DhobaleVishal) [inline src/commons/utils/corio_utils.py]: keep command in commands.py and then use it here would help if we need it for different package.

File: src/commons/utils/nimon_resource_util.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+from src.commons.utils.corio_utils import is_package_installed_local
+from src.commons.utils.cluster_utils import RemoteHost
+from src.commons import constants as cm_cmd
+from src.commons.constants import ROOT
+from config import CLUSTER_CFG
+
+LOGGER = logging.getLogger(ROOT)
+
+
+# pylint: disable-msg=too-many-statements
```
Comment (DhobaleVishal) [inline src/commons/utils/nimon_resource_util.py:33]: fix this codacy issue.

**Review by DhobaleVishal — CHANGES_REQUESTED** [review]

File: src/commons/constants.py (hunk 3 above)
Comment (akshaym99) [reply src/commons/constants.py]: Grouped & moved to commands.py

File: src/commons/utils/corio_utils.py (hunk 4 above)
Comment (akshaym99) [reply src/commons/utils/corio_utils.py]: moved commands to commands.py

File: src/commons/utils/nimon_resource_util.py (hunk 5 above)
Comment (akshaym99) [reply src/commons/utils/nimon_resource_util.py:33]: will fix this in Codacy war room

**Review by DhobaleVishal — APPROVED** [review]
@akshaym99 Attach execution logs.

**Review by ketanarlulkar — APPROVED** [review]

**Pull request merged**